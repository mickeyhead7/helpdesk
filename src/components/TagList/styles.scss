@import '../../themes/default/settings.media.scss';

.tagsList {
    .tagsHeader {
        .allTagsLink {
            display: inline-block;
            margin: 0 0 1em;
        }

        @media #{$desktopUp} {
            display: flex;
            align-items: center;

            .allTagsLink {
                margin: 0 0 1em auto;
            }
        }
    }

    .tagLinks {
        $tabletUp: "only screen and (min-width: 40.0625em)";
        $desktopUp: "only screen and (min-width: 60.0625em)";

        font-family: sans-serif;
        margin: 0 -1em 2em;

        @media #{$tabletUp} {
            display: flex;
            flex-wrap: wrap;
        }

        .tag {
            list-style: none;
            margin: 0 1em 1em;
            
            @media #{$tabletUp} {
                display: flex;
                flex-direction: column;
                width: calc(50% - 2em);
            }

            @media #{$desktopUp} {
                width: calc(25% - 2em);
            }
        }
    }
}
