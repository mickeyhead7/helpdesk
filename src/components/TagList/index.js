import React from 'react';
import Link from 'gatsby-link';
import propTypes from 'prop-types';

import './styles.scss';

const TagList = ({ tags }) => (
  <section className="tagsList">
    <header className="tagsHeader">
      <h2>Popular tags</h2>
      <Link className="allTagsLink" to="/tags">
        View all tags
      </Link>
    </header>
    <ul className="tagLinks">
      {tags.map((tag, index) =>
        <li className="tag" key={index}>
          <Link to={`/tags/${tag}`}>{tag}</Link>
        </li>
      )}
    </ul>
  </section>
);

TagList.propTypes = {
  tags: propTypes.array.isRequired,
};

export default TagList;
