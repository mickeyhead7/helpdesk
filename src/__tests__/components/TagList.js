import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Link from 'gatsby-link';
import TagList from '../../components/TagList';

const tags = ['tag-one', 'tag-two', 'tag-three'];

test('Header', () => {
  const component = shallow(<TagList tags={tags} />);
  const expectedTagsLength = tags.length + 1;

  expect(component.find(Link)).toHaveLength(expectedTagsLength);
  expect(toJson(component)).toMatchSnapshot();
});
