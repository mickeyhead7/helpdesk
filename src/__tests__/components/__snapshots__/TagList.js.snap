// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<section
  className="tagsList"
>
  <header
    className="tagsHeader"
  >
    <h2>
      Popular tags
    </h2>
    <Link
      className="allTagsLink"
      to="/tags"
    >
      View all tags
    </Link>
  </header>
  <ul
    className="tagLinks"
  >
    <li
      className="tag"
      key="tag-one"
    >
      <Link
        to="/tags/tag-one"
      >
        tag-one
      </Link>
    </li>
    <li
      className="tag"
      key="tag-two"
    >
      <Link
        to="/tags/tag-two"
      >
        tag-two
      </Link>
    </li>
    <li
      className="tag"
      key="tag-three"
    >
      <Link
        to="/tags/tag-three"
      >
        tag-three
      </Link>
    </li>
  </ul>
</section>
`;
